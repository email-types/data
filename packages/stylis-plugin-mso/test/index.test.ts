import stylis from 'stylis';
import plugin, { createPlugin, Plugin } from '../src';

const specs: Array<{
  name: string;
  sample: string;
  expected: string;
  prefix?: boolean;
}> = [
  {
    name: 'returns with mso prefix',
    sample: 'color: tomato;',
    expected: '{mso-color-alt:tomato;color:tomato;}',
    prefix: true,
  },
  {
    name: 'returns without mso prefix',
    sample: 'color: tomato;',
    expected: '{color:tomato;}',
    prefix: false,
  },
  {
    name: 'returns with a fixed mso prefix',
    sample: '-mso-color-alt: tomato;',
    expected: '{mso-color-alt:tomato;}',
  },
];

describe('stylis-plugin-mso', () => {
  specs.forEach((spec) => {
    it(spec.name, () => {
      let _plugin: Plugin = plugin;

      if (typeof spec.prefix === 'boolean') {
        stylis.set({ prefix: spec.prefix });

        _plugin = spec.prefix
          ? createPlugin()
          : createPlugin({ prefix: spec.prefix });
      }

      stylis.use(null)(_plugin);

      const result = stylis('', spec.sample);
      expect(result).toMatchSnapshot();
      expect(result).toEqual(spec.expected);
    });
  });
});
