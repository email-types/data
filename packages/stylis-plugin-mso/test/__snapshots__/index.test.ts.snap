// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stylis-plugin-mso returns with a fixed mso prefix 1`] = `"{mso-color-alt:tomato;}"`;

exports[`stylis-plugin-mso returns with mso prefix 1`] = `"{mso-color-alt:tomato;color:tomato;}"`;

exports[`stylis-plugin-mso returns without mso prefix 1`] = `"{color:tomato;}"`;
